javagen/src/main/java/com/azure/autorest/template/ProtocolSampleTemplate.java

```diff
@@ -98,10 +103,33 @@ public void write(ProtocolExample protocolExample, JavaFile javaFile) {
                 method.getProxyMethod().getAllParameters().stream().filter(p -> !p.getFromClient()).filter(p -> p.getName().equalsIgnoreCase(parameterName)).findFirst().ifPresent(p -> {
                     switch (p.getRequestParameterLocation()) {
                         case Query:
-                            requestOptionsStmts.add(
-                                    String.format("requestOptions.addQueryParam(\"%s\", %s);",
-                                            parameterName,
-                                            p.getClientType().defaultValueExpression(parameterValue.getUnescapedQueryValue().toString())));
+                            if (parameterValue.getUnescapedQueryValue() instanceof List && p.getCollectionFormat() != null) {
```

Comment:
Handle example for `multi`, `csv`, etc. for query with array type and `collectionFormat`.